packages/front-end/components/Experiment/NewExperimentForm.tsx

```diff
@@ -387,6 +388,7 @@ const NewExperimentForm: FC<NewExperimentFormProps> = ({
                   value: v.key || "",
                   name: v.name,
                   weight: form.watch(`phases.0.variationWeights.${i}`),
+                  id: v.id || "",
```

Comment:
These too - we can do without

Reply:
I think we do need to retain this. When creating an experiment, the shape of a variation is slightly different than when we create an override rule for a feature flag. Confusingly, when creating a new experiement, the shape of a variation is `Variation`, and when creating a variation via the override rule modal, it's `ExperimentValue`.

And, when creating an experiment, in order to pass the variation into the `VariationsInput` component, we're converting the experiment variation into a `Variation`. (Essentially removing properties like `screenshots`, `css`, `dom`, and `key`.) If we remove this, the id get's wiped, so the work we're doing within `VariationsInput` that adds an id to each variation if one doesn't exist, has to run each time, causing a flicker.

Reply:
Thank you for explaining, I see your point. It makes sense to leave this. 

My previous suggestions to remove the `id` field from back-end types may make this difficult.

You can certainly leave the types as-is, though like you mentioned earlier, it is a bit hacky.

One way you could 'contain' the hackiness is to define the `id` field in the type definition inside `NewExperiementForm` by augmenting `ExperimentInterfaceStringDates` (used to define the `initialValue` prop) to contain Variations of type `DraggableVariations`, for example.